File: python/jieba/performance.py
#encoding=utf-8
import sys
sys.path.append("/Users/yanyiwu/code/jieba/")
import jieba
import random
import datetime

if __name__ == "__main__":
    lines = []
    for line in open("/Users/yanyiwu/code/practice/nodejs/nodejieba/performance/weicheng.utf8"):
        lines.append(line.strip());


    result = [""] * 10;
    result[random.randint(0, 9)] = '/'.join(jieba.cut("南京长江大桥"))
    starttime = datetime.datetime.now()
    print >> sys.stderr, starttime
    for i in xrange(50):
        for line in lines:
            result[random.randint(0, 9)] = '/'.join(jieba.cut(line))
            #result[random.randint(0, 9)] = jieba.cut(line)
    endtime = datetime.datetime.now()
    print endtime

File: go/gojieba/performance/performance.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"time"

	"github.com/yanyiwu/gojieba"
)

func main() {
	jieba := gojieba.NewJieba("./jieba.dict.utf8", "./hmm_model.utf8", "user.dict.utf8")
	defer jieba.Free()
	lines := []string{}
	f, err := os.Open("../../../nodejs/nodejieba/performance/weicheng.utf8")
	defer f.Close()
	if nil == err {
		buff := bufio.NewReader(f)
		for {
			line, err := buff.ReadString('\n')
			if err != nil || io.EOF == err {
				break
			}
			lines = append(lines, line)
		}
	}
	fmt.Println(len(lines))
	fmt.Println(time.Now())
	for i := 0; i < 50; i++ {
		for j := 0; j < len(lines); j++ {
			jieba.Cut(lines[j], true)
		}
	}
	fmt.Println(time.Now())
}
